Lib/fontTools/varLib/instancer.py

```diff
@@ -0,0 +1,1031 @@
+with the rest planned to be implemented in the future, namely:
+L1) dropping one or more axes while leaving the default tables unmodified;
+L2) dropping one or more axes while pinning them at non-default locations;
+L3) restricting the range of variation of one or more axes, by setting either
+    a new minimum or maximum, potentially -- though not necessarily -- dropping
+    entire regions of variations that fall completely outside this new range.
+L4) moving the default location of an axis.
+
+Currently only TrueType-flavored variable fonts (i.e. containing 'glyf' table)
+are supported, but support for CFF2 variable fonts will be added soon.
+
+The discussion and implementation of these features are tracked at
+https://github.com/fonttools/fonttools/issues/1537
+
+NOTE: The module is experimental and both the API and the CLI *may* change.
+"""
+from __future__ import print_function, division, absolute_import
+from fontTools.misc.py23 import *
+from fontTools.misc.fixedTools import floatToFixedToFloat, otRound
+from fontTools.varLib.models import supportScalar, normalizeValue, piecewiseLinearMap
+from fontTools.ttLib import TTFont
+from fontTools.ttLib.tables.TupleVariation import TupleVariation
+from fontTools.ttLib.tables import _g_l_y_f
+from fontTools import varLib
+
+# we import the `subset` module because we use the `prune_lookups` method on the GSUB
+# table class, and that method is only defined dynamically upon importing `subset`
+from fontTools import subset  # noqa: F401
+from fontTools.varLib import builder
+from fontTools.varLib.mvar import MVAR_ENTRIES
+from fontTools.varLib.merger import MutatorMerger
+from contextlib import contextmanager
+import collections
+from copy import deepcopy
+import logging
+from itertools import islice
+import os
+import re
+
+
+log = logging.getLogger("fontTools.varLib.instancer")
+
+
+def instantiateTupleVariationStore(variations, location, origCoords=None, endPts=None):
+    newVariations = collections.OrderedDict()
+    for var in variations:
+        # Compute the scalar support of the axes to be pinned at the desired location,
+        # excluding any axes that we are not pinning.
+        # If a TupleVariation doesn't mention an axis, it implies that the axis peak
+        # is 0 (i.e. the axis does not participate).
+        support = {axis: var.axes.pop(axis, (-1, 0, +1)) for axis in location}
+        scalar = supportScalar(location, support)
+        if scalar == 0.0:
+            # no influence, drop the TupleVariation
+            continue
+
+        # compute inferred deltas only for gvar ('origCoords' is None for cvar)
+        if origCoords is not None:
+            var.calcInferredDeltas(origCoords, endPts)
+
+        var.scaleDeltas(scalar)
+
+        # merge TupleVariations with overlapping "tents"
+        axes = tuple(var.axes.items())
+        if axes in newVariations:
+            newVariations[axes] += var
+        else:
+            newVariations[axes] = var
```

Comment:
get w/default, no if?

Reply:
I'm setting, not getting. Maybe you mean `setdefault` method?
Here I'm grouping the TupleVariations that have the same axes (same axis tags and and same (min/default/max)) and summing their deltas as I go.